### PR TITLE
[Halide] Fix download url

### LINKED ISSUE #42269: [halide] Build error on x64-windows
Package: halide[core,target-x86]:x64-windows@18.0.0

**Host Environment**

- Host: x64-windows
- Compiler:  
-    vcpkg-tool version: 2024-11-12-eb492805e92a2c14a230f5c3deb3e89f6771c321
    vcpkg-scripts version: 834977918 2024-11-20 (6 hours ago)

**To Reproduce**

`vcpkg install halide:x64-windows`

**Failure logs**

```
Downloading halide-Halide-18.0.0.tar.gz
Failed to download halide-Halide-18.0.0.tar.gz.
If you are using a proxy, please ensure your proxy settings are correct.
Possible causes are:
1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable to `https//address:port`.
This is not correct, because `https://` prefix claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr, etc...) is an HTTP proxy.
Try setting `http://address:port` to both HTTP_PROXY and HTTPS_PROXY instead.
2. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings set by your proxy software. See, https://github.com/microsoft/vcpkg-tool/pull/77
The value set by your proxy might be wrong, or have same `https://` prefix issue.
3. Your proxy's remote server is our of service.
If you've tried directly download the link, and believe this is not a temporay download server failure, please submit an issue at https://github.com/Microsoft/vcpkg/issues
to report this upstream download server failure.
error: https://github.com/halide/Halide/archive/18.0.0.tar.gz: failed: status code 404
CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:231 (message):
  Download failed, halting portfile.
Call Stack (most recent call first):
  scripts/cmake/vcpkg_from_github.cmake:106 (vcpkg_download_distfile)
  ports/halide/portfile.cmake:3 (vcpkg_from_github)
  scripts/ports.cmake:196 (include)

### PR DESCRIPTION
**Pull request by FrankXie05**
Fix https://github.com/microsoft/vcpkg/issues/42269

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version.~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [ ] ~Any patches that are no longer applied are deleted from the port's directory.~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by LilyWangLL — DISMISSED** [review]

File: ports/halide/portfile.cmake [hunk 1]
```diff
@@ -3,7 +3,7 @@ vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO halide/Halide
-    REF "${VERSION}"
+    REF "v${VERSION}"
     SHA512 286cbef25b5cc0f5095cbc80a2fd1cacf369948c58c14406ac6bcc28a7a37c81417d601975083f03670e22276a1886b8801bdc91aa6fe80049a276a1c8fd08b9
```
Comment (BillyONeal) [inline ports/halide/portfile.cmake:7]: The SHA is almost certainly wrong. I'm checking....

Comment (FrankXie05) [reply ports/halide/portfile.cmake]: Weirdly, this doesn't trigger CI.

Comment (dg0yt) [reply ports/halide/portfile.cmake]: Not weird. 
Asset caching has a match for the old SHA512, so a new download is not attempted.
Causing bad merges several times a year when reviewers don't pay attention.

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**